scripts/storage-emulator-integration/rules/manager.test.ts

```diff
@@ -1,92 +1,108 @@
 import { expect } from "chai";
-import { tmpdir } from "os";
-import { v4 as uuidv4 } from "uuid";
 
-import { FirebaseError } from "../../../src/error";
-import { StorageRulesFiles, TIMEOUT_MED } from "../../../src/test/emulators/fixtures";
-import { StorageRulesManager } from "../../../src/emulator/storage/rules/manager";
+import {
+  createTmpDir,
+  StorageRulesFiles,
+  TIMEOUT_LONG,
+} from "../../../src/test/emulators/fixtures";
+import {
+  createStorageRulesManager,
+  StorageRulesManager,
+} from "../../../src/emulator/storage/rules/manager";
 import { StorageRulesRuntime } from "../../../src/emulator/storage/rules/runtime";
 import { Persistence } from "../../../src/emulator/storage/persistence";
-import { RulesetOperationMethod } from "../../../src/emulator/storage/rules/types";
+import { RulesetOperationMethod, SourceFile } from "../../../src/emulator/storage/rules/types";
+import { isPermitted } from "../../../src/emulator/storage/rules/utils";
+import { readFile } from "../../../src/fsutils";
 
 describe("Storage Rules Manager", function () {
   const rulesRuntime = new StorageRulesRuntime();
-  const rulesManager = new StorageRulesManager(rulesRuntime);
+  const rules = [
+    { resource: "bucket_0", rules: StorageRulesFiles.readWriteIfTrue },
+    { resource: "bucket_1", rules: StorageRulesFiles.readWriteIfAuth },
+  ];
+  const opts = { method: RulesetOperationMethod.GET, file: {}, path: "/b/bucket_2/o/" };
+  let rulesManager: StorageRulesManager;
 
   // eslint-disable-next-line @typescript-eslint/no-invalid-this
-  this.timeout(TIMEOUT_MED);
+  this.timeout(TIMEOUT_LONG);
 
-  before(async () => {
+  beforeEach(async () => {
     await rulesRuntime.start();
+
+    rulesManager = createStorageRulesManager(rules, rulesRuntime);
+    await rulesManager.start();
   });
 
-  after(async () => {
+  afterEach(async () => {
     rulesRuntime.stop();
-    await rulesManager.close();
+    await rulesManager.stop();
   });
 
-  it("should load ruleset from SourceFile object", async () => {
-    await rulesManager.setSourceFile(StorageRulesFiles.readWriteIfTrue);
-    expect(rulesManager.ruleset).not.to.be.undefined;
+  it("should load multiple rulesets on start", async () => {
```

Comment:
do we have a test for the multiple target case failing to find the security rules for a resource?